### PR TITLE
POR-2115 improve app creation in new update flow

### PR DESCRIPTION
**Pull request by ianedwards**
## What does this PR do?

- Properly set source type when creating an app, which has previously overridden by docker repo regardless of type
- skip validate call when running update flow
- show that app is awaiting build when the hello world image is being used

### CODE REVIEW

File: api/server/handlers/porter_app/update_app.go [hunk 1]
```diff
@@ -253,16 +259,33 @@ func (c *UpdateAppHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	c.WriteResult(w, r, response)
 }
 
-func sourceFromAppAndGitSource(appProto *porterv1.PorterApp, gitSource GitSource) (porter_app.SourceType, *porter_app.Image) {
+func sourceFromAppAndGitSource(ctx context.Context, appProto *porterv1.PorterApp, gitSource GitSource) (porter_app.SourceType, *porter_app.Image, error) {
```
Comment (ferozemohideen) [inline api/server/handlers/porter_app/update_app.go:262]: any reason this has to take in a proto? why not nil check in the caller function?

Comment (ianedwards) [reply api/server/handlers/porter_app/update_app.go:262]: I don't have a strong preference. / reason one way or the other, but proto objects we've always kept as pointers

File: dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx [hunk 2]
```diff
@@ -236,6 +238,15 @@ const AppDataContainer: React.FC<AppDataContainerProps> = ({ tabParam }) => {
             variables,
             secrets,
             is_env_override: true,
+            deletions: {
+              service_names: data.deletions.serviceNames.map((s) => s.name),
+              predeploy: data.deletions.predeploy.map((s) => s.name),
+              env_group_names: data.deletions.envGroupNames.map(
+                (eg) => eg.name
+              ),
+              env_variable_names: [],
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx:247]: do we still need to pass this in?

File: dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/DeployEventCard.tsx [hunk 3]
```diff
@@ -276,14 +276,14 @@ const DeployEventCard: React.FC<Props> = ({
                 </Link>
               </ImageTagContainer>
             </>
-          ) : (
+          ) : event.metadata.image_tag ? (
             <>
               <Spacer inline x={0.5} />
```
Comment (ferozemohideen) [inline dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/DeployEventCard.tsx:281]: can you add the tag icon while you're here?

File: dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx (hunk 2 above)
Comment (ferozemohideen) [reply dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx:247]: empty env_variable_names i mean

**Review by ferozemohideen — APPROVED** [review]

File: dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx (hunk 2 above)
Comment (ianedwards) [reply dashboard/src/main/home/app-dashboard/app-view/AppDataContainer.tsx:247]: i can go ahead and make it optional at this point

File: dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/DeployEventCard.tsx (hunk 3 above)
Comment (ianedwards) [reply dashboard/src/main/home/app-dashboard/app-view/tabs/activity-feed/events/cards/DeployEventCard.tsx:281]: sure thing

**Pull request merged**